### DISCUSSION TITLE
Add a `withSchedule` method to the `ApplicationBuilder` class

### DISCUSSION THREAD

**Discussion by martinbean** [laravel/framework#50493] [Ideas] [answered]
I’m exploring Laravel 11. One thing I don’t really like is defining scheduled tasks in a **routes/console.php** file. I’ve never liked the thought of “routes” for CLI.

I see a lot of stuff has been moved to the `ApplicationBuilder` class called in the **bootstrap/app.php** file. It would be nice if this class could expose a `withSchedule` method to allow registering of scheduled tasks in lieu of the now-removed console kernel file.

In my mind, it would look something like this:

```php
return Application::configure(basePath: dirname(__DIR__))
    // ...
    ->withSchedule(function (Schedule $schedule) {
        $schedule->command('inspire')->hourly();
    })
    ->create();
```

Thoughts? I’m also happy to put the PR together if this was wanted.

**Comment by adevade** [comment]
Yeah, I like this idea! 👍

**Comment by martinbean** [comment]
@taylorotwell @driesvints Any thoughts on this? Given the number of up-votes this has garnered versus other ideas, it seems I’m not alone in wanting this feature.

**Comment by nunomaduro** [comment, marked as answer]
Done here: https://github.com/laravel/framework/pull/50755. It's a great suggestion.

**Reply by martinbean** [reply]
Awesome. Thanks, @nunomaduro 🙂
